src/subsystems/nodejs/builders/granular/default.nix

```diff
@@ -148,13 +158,34 @@
 
     # Generates a derivation for a specific package name + version
     makePackage = name: version: let
+      pname = lib.replaceStrings ["@" "/"] ["__at__" "__slash__"] name;
+
       deps = getDependencies name version;
 
       nodeDeps =
         lib.forEach
         deps
         (dep: allPackages."${dep.name}"."${dep.version}");
 
+      nodeModulesDir = pkgs.runCommand "node_modules-${pname}" {} ''
+        mkdir $out
+        ${l.concatStringsSep "\n" (
+          l.forEach nodeDeps
+          (pkg: ''
+            ln -s ${pkg}/lib/node_modules/* $out/
+          '')
+        )}
+      '';
+
+      nodeModulesBinDirs = pkgs.runCommand "node_modules_bin-${pname}" {} ''
+        touch $out
+        for dep in ${l.toString nodeDeps}; do
+          for bin in $(${pkgs.findutils}/bin/find $dep -name .bin); do
```

Comment:
Isn't there only one .bin folder at the top of the node_modules? so compose .bin in the node_modules directory with something like
```sh
if [ -d ${dep}/lib/node_modules/.bin ]; then
  ln -s ${dep}/lib/node_modules/.bin/* $out/.bin
fi
```
?